pkg/ast/spl/spl.peg

```diff
@@ -5203,8 +5221,148 @@ MvexpandLimit <- SPACE "limit" EQUAL intValue:IntegerAsString {
     return intValue, nil
 }
 
+OptionalWhereClause <- SPACE CMD_WHERE condition:BoolExpr {
+    return condition.(*structs.BoolExpr), nil
+}
```

Comment:
Marked this rule as optional because it could be used at other places too in the future where we need an optional where clause just like how it is used in the input lookup.

Reply:
Ok but in other rules you use it like:
```
whereClause:(OptionalWhereClause)?
```
In those rules, the `?` is what makes it optional. I think naming this `OptionalWhereClause` is misleading because there's nothing inherently optional about the rule; it's just how you're using it in other rules that makes it optional

Reply:
addressed in the latest commit.